delivery/operator/README.md

```diff
@@ -0,0 +1,260 @@
+resources, then declaring such a resource per the following instructions. You
+may also want to review the test script at
+[./test/build-and-push-operator.sh](./test/build-and-push-operator.sh).
+
+### Build a Helm chart-based operator
+
+In an empty directory init a new Helm operator from the podtato-head chart as follows.
+
+```bash
+chart_dir=./delivery/chart
+# expand relative path so it works from other dirs
+chart_dir=$(cd ${chart_dir} && pwd)
+
+work_dir=./work
+mkdir -p ${work_dir}
+cd ${work_dir}
+
+operator-sdk init --plugins 'helm.sdk.operatorframework.io/v1' --project-version=3 \
+    --project-name 'podtato-head-operator' \
+    --domain=podtato-head.io \
+    --group=apps \
+    --version=v1alpha1 \
+    --kind=PodtatoHeadApp \
+    --helm-chart=${chart_dir}
+```
+
+Now build and push the operator image, the operator bundle, and a catalog which
+includes the bundle as follows. We use GitHub's container registry `ghcr.io`;
+you will need to be signed in to push images. Sign in by getting a personal
+access token with `packages:write` permissions and running the following commands.
+
+```bash
+github_user=
+github_token=
+
+echo "${github_token}" | docker login --username=${github_user} --password-stdin ghcr.io
+
+export IMAGE_TAG_BASE=ghcr.io/${github_user}/podtato-head/operator
+export IMG=${IMAGE_TAG_BASE}:latest
+
+make docker-build docker-push
+make bundle bundle-build bundle-push
+make catalog-build catalog-push
+```
+
+### Deploy the operator
+
+To deploy the operator we'll deploy the catalog source and then subscribe to our
+package, as follows. You may also want to review the test script at
+[./test/deploy-operator.sh](./test/deploy-operator.sh).
+
+First deploy the catalog source to the OLM namespace:
+
+```bash
+image_base_url=ghcr.io/${github_user}/podtato-head/operator
+operator_version=0.0.1
+
+kubectl apply -f - <<EOF
+apiVersion: operators.coreos.com/v1alpha1
+kind: CatalogSource
+metadata:
+    name: podtato-head-catalog
+    namespace: olm
+spec:
+    sourceType: grpc
+    image: ${image_base_url}-catalog:v${operator_version}
+    secrets:
+      - ghcr
+    displayName: podtato-head-catalog
+    publisher: podtato-head
+    updateStrategy:
+        registryPoll:
+            interval: 10m
+EOF
+```
+
+Next declare a subscription to the podtato-head-operator in the `operators` namespace:
+
+```bash
+operator_version=0.0.1
+
+kubectl apply -f - <<EOF
+apiVersion: operators.coreos.com/v1alpha1
+kind: Subscription
+metadata:
+    name: podtato-head-operator
+    namespace: operators
+spec:
+    channel: alpha
+    installPlanApproval: Automatic
+    name: podtato-head-operator
+    source: podtato-head-catalog
+    sourceNamespace: olm
+    startingCSV: podtato-head-operator.v${operator_version}
+EOF
+```
+
+The subscription triggers creation of a ClusterServiceVersion (CSV) resource its
+namespace, followed by an InstallPlan that deploys Deployments, Service
+Accounts, and other resources associated with the operator.
```

Comment:
This is slightly off. Creating a Subscription causes the Catalog Operator to generate an installplan which records a list of resources that need to be installed in the `status.plan` array, the CSV is included in this list. The OLM operator will then reconcile the InstallPlan, creating each resource listed.